File: 1220-count-vowels-permutation/1220-count-vowels-permutation.cpp
class Solution {
public:
    int countVowelPermutation(int n) {
        vector<long> prev(5,1);
            
        const int mod = 1e9 + 7;
        
        long ans = 5;
        
        for(int i=1;i<n;i++)
        {
            vector<long> curr(5);
            
            curr[0] = (prev[1] + prev[2] + prev[4])%mod;
            curr[1] = (prev[0] + prev[2])%mod;
            curr[2] = (prev[1] + prev[3])%mod;
            curr[3] = (prev[2])%mod;
            curr[4] = (prev[2] + prev[3])%mod;
            
            prev = curr;
        }
        
        return accumulate(prev.begin(),prev.end(),0LL)%mod;        
    }
};

File: 1220-count-vowels-permutation/1220-count-vowels-permutation.py
class Solution:
    def countVowelPermutation(self, n: int) -> int:
        a,e,i,o,u = 1,1,1,1,1
        mod = 1e9 + 7
        for _ in range(1,n):
            na,ne,ni,no,nu = (e+i+u)%mod , (a+i)%mod , (o+e)%mod, (i)%mod , (i+o)%mod
            a,e,i,o,u = na,ne,ni,no,nu
        return int((a+e+i+o+u)%mod)